lib/device_set.js

```diff
@@ -37,30 +38,86 @@ function DeviceSetModule(config, log, homebridge, vivintApi, ThermostatCharacter
       let device = new deviceClass(accessory)
       this.devices.push(device)
       this.devicesById[device.id] = device
+
+      if (accessory.context.deviceClassName === "Panel") this.panel_Id = device.id
     }
 
     handleSnapshot(deviceData, timestamp) {
       this.lastSnapshotTime = timestamp
       log("Handling incoming device snapshot for timestamp", timestamp)
+
       for (let _deviceId in this.devicesById) {
         let deviceId = parseInt(_deviceId)
         let d = deviceData.find((dvc) => dvc._id == deviceId)
         if (d) this.devicesById[_deviceId].handleSnapshot(d)
       }
+      
+      
+      // update armed status
+      let m = {}
+      m.message = {
+      	 _id: this.panel_Id,
+         s: vivintApi.systemInfo.system.s, 
+         stateupdate: true,
+         newstate: vivintApi.systemInfo.system.s,
+         da: {
+	         plctx: {
+		         ts: 0
+	         }
+         }
+      }
+
+      if (vivintApi.systemInfo.system.s === 0) m.message.da.secd = { s: vivintApi.systemInfo.system.s }
+      else m.message.da.seca = { s: vivintApi.systemInfo.system.s }
+    
+
+      this.handleMessage(m)  
+      
+      // end update arm status    
     }
 
     handleMessage(msg) {
-      if ((msg.message.da) && (msg.message.da.d)) {
-        if (msg.message.da.plctx.ts < this.lastSnapshotTime) {
-          log("Ignoring stale update", msg.message.da.plctx.ts, "<", this.lastSnapshotTime)
-          return;
-        }
-        msg.message.da.d.forEach((patch) => {
-          if (this.devicesById[patch._id]) {
-            this.devicesById[patch._id].handlePatch(patch)
-          }
-        })
-      }
+     
+     if (msg.message.da.seca) {
```

Comment:
can you please extract this out to a function? We should keep this method simple.

Also, I wonder if we should process ARM / DISARM messages if the update is stale (we can receive messages that are older than the snapshot we achieved during initial auth and periodic re-auth)